posts/auditing-using-javers.md

```diff
@@ -0,0 +1,238 @@
+
+The usage of this tool is not limited to debugging and auditing only. It can be successfully applied to perform analysis, force security policies and maintaining the event log.
+
+# Spring Boot Integration and Setup
+JaVers provide a spring boot starter that integrate all the required JaVers beans with default configuration.
+
+pom.xml
+```xml
+<dependency>
+			<groupId>org.javers</groupId>
+			<artifactId>javers-spring-boot-starter-mongo</artifactId>
+			<version>5.8.8</version>
+</dependency>
+```
+
+
+# JaVers repository configuration
+JaVers spring boot starter is configured to reuse the application's database (managed by Spring Data).
+Alternativelly, we can choose to store JaVers data in a separate database as shown bellow:
+
+```yaml
+javers:
+  mongodb:
+    host: localhost
+    port: 27017
+    database: javers-audit
+    authentication-database: admin
+    username: javers
+    password: password
+```
+Or
+
+```yaml
+javers:
+  mongodb:
+    uri: mongodb://javers:password@localhost:27017/javers-audit&authSource=admin
+```
+
+# Auto audit annotations
+
+Auditing specific entities can be done very easy without additionnal code, simply add the @JaversSpringDataAuditable on the corresponding Spring Data repositories.
+
+Example on MongoRepository
+```
+import org.javers.spring.annotation.JaversSpringDataAuditable
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.javers.organization.structure.domain.Person;
+
+@JaversSpringDataAuditable
+public interface PersonRepository extends MongoRepository<Person, String> {
+}
+```
+All the changes made to Person objects will be audited by JaVers.
+
+In case of a custom repository, the @JaversAuditable needs to be added at the method level:
+
+```java
+@Repository
+class UserRepository {
+
+    @JaversAuditable
+    public void save(User user) {
+        ...//
+    }
+
+    public User find(String login) {
+        ...//
+    }
+}
+```
+
+For a fine-grained control, the JaVers spring bean can be used:
+
+```java
+@Service
+class PersonService {
+    private final Javers javers;
+    private final PersonRepository personRepository;
+
+    @Autowired
+    public PersonService(Javers javers, PersonRepository personRepository) {
+        this.javers = javers;
+        this.personRepository = personRepository;
+    }
+
+    public void fire(Person person) {
+        person.fire();
+        personRepository.save(person);
+
+        if (LocalDate.now().getDayOfWeek() == DayOfWeek.FRIDAY){
+            javers.commit("author", person);
+        }
+    }
+}
+```
+In this example the auditing is only recorded on Fridays.
+
+# Domain model annotations
+One of the challenges faced in our client is to keep the size of the audited data reasonable. JaVers provide a good domain model configuration, and we were able to use the @ValueObject annotation at the class level to audit on our embedded objects
```

Comment:
```suggestion
One of the challenges faced in our client is to keep the size of the audited data reasonable. JaVers provides a good domain model configuration. We were able to use the `@ValueObject` annotation at the class level to audit our embedded objects.
```

Reply:
"One of the challenges our client faced was keeping the size of the audited data reasonable."